robot_calibration/src/calibrate.cpp

```diff
@@ -64,6 +65,126 @@
  *     - write results to URDF.
  */
 
+void output_calibration_offsets(const robot_calibration::OptimizationParams& params,
+                                const robot_calibration::Optimizer& opt,
+                                const std::vector<robot_calibration_msgs::CalibrationData>& data,
+                                const std_msgs::String& description_msg, const ros::NodeHandle& nh)
+{
+  // Generate datecode
+  char datecode[80];
+  {
+    std::time_t t = std::time(NULL);
+    std::strftime(datecode, 80, "%Y_%m_%d_%H_%M_%S", std::localtime(&t));
+  }
+
+  std::string output_directory;
+  if (!nh.getParam("output_directory", output_directory))
+  {
+    output_directory = "/tmp/calibration/" + std::string(datecode);
+  }
+
+  boost::filesystem::path dir(output_directory);
+
+  if (!boost::filesystem::exists(dir))
+  {
+    boost::filesystem::create_directories(dir);
+  }
+
+  ROS_INFO("saving offsets to yaml");
+  {
+    std::stringstream stream;
+
+    stream << "frames_offsets:" << std::endl;
+    for (uint32_t i = 0U; i < params.free_frames.size(); i++)
```

Comment:
why don't you use `size_t` in for loops?
if you chose to, there is plenty of loops to apply this change...

Reply:
no specific reason for 32bit, only unsigned was intended. Changed it at locations where it makes sense to ```size_t```